src/Mod/Part/App/TopoShape.cpp

```diff
@@ -2431,16 +2431,14 @@ TopoDS_Shape TopoShape::makeLongHelix(Standard_Real pitch, Standard_Real height,
     Handle(Geom_Surface) surf;
     Standard_Boolean isCylinder;
 
-    if (angle < Precision::Confusion()) {                                      // Cylindrical helix
+    if (angle < Precision::Confusion() && angle > -Precision::Confusion()) {                                      // Cylindrical helix

```

Comment:
This should be changed to `if (std::fabs(angle) < Precision::Confusion())` so its meaning is more obvious

Reply:
Good point. Done